### PR TITLE
Improve HTTP::Cookies

### PR DESCRIPTION
**Pull request by jhass**
Mainly handles the points raised in https://github.com/manastech/crystal/pull/1252#issuecomment-140660698, also adds docs and some convenience API.

### CODE REVIEW

File: src/http/request.cr [hunk 1]
```diff
@@ -16,6 +16,8 @@ class HTTP::Request
     end
   end
 
+  # Returns a convenience wrapper around querying and setting cookie related
+  # headers, see *HTTP::Cookies*
```
Comment (asterite) [inline src/http/request.cr]: To automatically link to other types use backticks

Comment (jhass) [inline src/http/request.cr]: Ah, not too intuitive tbh, since it's also used for inline code. I prefer YARD's `{Foo}`.

Comment (netfeed) [inline src/http/request.cr]: That's just me not knowing the language well enough.

**Comment by jhass** [comment]
Whoops, fixed the typo

**Comment by asterite** [comment]
@jhass Thanks! I think we are just missing `HTTP::Cookies#[]?`. I also think we can make that type a struct, it's just a wrapper around a Hash.

**Pull request merged**